File: desktop/src/main/java/com/corrodinggames/rts/game/FrameBufferHelper.java
package com.corrodinggames.rts.game;

import android.graphics.Paint;
import android.graphics.Rect;
import com.corrodinggames.rts.gameFramework.AssetType;
import com.corrodinggames.rts.gameFramework.GameEngine;
import com.corrodinggames.rts.gameFramework.graphics.GamePaint;
import com.corrodinggames.rts.gameFramework.graphics.GraphicsEngine;
import com.corrodinggames.rts.gameFramework.graphics.ShaderProgram;
import com.corrodinggames.rts.gameFramework.graphics.Texture;
import java.io.IOException;

/* JADX INFO: renamed from: com.corrodinggames.rts.game.j */
/* JADX INFO: loaded from: game-lib.jar:com/corrodinggames/rts/game/j.class */
public class FrameBufferHelper {
    public Texture a;
    GraphicsEngine b;
    GamePaint c;
    ShaderProgram d;
    Paint e;
    Rect f;
    boolean g;

    public FrameBufferHelper() {
        this.e = new Paint();
        this.f = new Rect(-101, 0, -1, 100);
        this.c = new GamePaint();
    }

    public FrameBufferHelper(String str) {
        this();
        try {
            this.d = new ShaderProgram(str);
            this.c.a(this.d);
            if (this.d.o != 0) {
                this.g = true;
            }
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }

    public boolean a() {
        if ((this.d != null && this.d.o != 0) || this.g) {
            return true;
        }
        return false;
    }

    public void a(GraphicsEngine graphicsEngine) {
        a(graphicsEngine, graphicsEngine.m(), graphicsEngine.n(), 10);
    }

    public void a(GraphicsEngine graphicsEngine, int i, int i2, int i3) {
        if (this.g) {
            return;
        }
        if (this.a != null && (i > this.a.m() || i2 > this.a.l())) {
            this.a.o();
            this.a = null;
            this.b = null;
        }
        if (this.a == null) {
            try {
                this.a = graphicsEngine.a(i + i3, i2 + i3, true);
                this.b = graphicsEngine.a(this.a);
            } catch (OutOfMemoryError e) {
                this.g = true;
                GameEngine.reportOOM(AssetType.gameImageCreate, e);
                return;
            }
        }
        this.b.a(i, i2);
    }

    public void b() {
        GameEngine gameEngine = GameEngine.getInstance();
        gameEngine.renderGraphicsEngine.b(this.f, this.e);
        gameEngine.renderGraphicsEngine.b(this.a, 0.0f, 0.0f, this.c);
    }
}

File: core/src/main/kotlin/com/corrodinggames/rts/game/FrameBufferHelper.kt
package com.corrodinggames.rts.game

import com.corrodinggames.rts.gameFramework.AssetType
import com.corrodinggames.rts.gameFramework.GameEngine
import com.corrodinggames.rts.gameFramework.graphics.*
import io.github.rwx.geometry.Rect
import io.github.rwx.render.canvas.KoolDisplacementEffect
import io.github.rwx.render.canvas.KoolPaint

class FrameBufferHelper() {
    private var graphicsBackend: GraphicsEngine? = null

    @JvmField
    var a: Texture? = null

    @JvmField
    var b: GraphicsEngine? = null

    @JvmField
    val c: GamePaint = GamePaint()

    @JvmField
    var d: KoolDisplacementEffect? = null

    @JvmField
    var shaderProgram: ShaderProgram? = null

    @JvmField
    val e: KoolPaint = KoolPaint()

    @JvmField
    val f: Rect = Rect(-101, 0, -1, 100)

    @JvmField
    var g: Boolean = false

    constructor(shaderPath: String) : this() {
        shaderProgram = ShaderProgram(shaderPath)
        c.a(shaderProgram)
        if (shaderProgram?.o != 0) {
            g = true
        }
    }

    fun a(): Boolean = g || (shaderProgram?.o ?: 0) != 0

    fun a(graphicsEngine: GraphicsEngine) {
        a(graphicsEngine, graphicsEngine.m(), graphicsEngine.n(), 10)
    }

    fun a(graphicsEngine: GraphicsEngine, width: Int, height: Int, padding: Int) {
        if (g) {
            return
        }
        if (graphicsBackend !== graphicsEngine) {
            releaseTarget()
            graphicsBackend = graphicsEngine
        }
        if (a != null && (width > a!!.width() || height > a!!.height())) {
            releaseTarget()
        }
        if (a == null) {
            try {
                a = graphicsEngine.a(width + padding, height + padding, true)
                b = graphicsEngine.b(a, RenderTargetMode.DEFAULT)
            } catch (e: OutOfMemoryError) {
                g = true
                GameEngine.reportOOM(AssetType.gameImageCreate, e)
                return
            }
        }
        b!!.a(width, height)
    }

    private fun releaseTarget() {
        b?.q()
        b = null
        a?.releaseTexture()
        a = null
    }

    fun b() {
        val gameEngine = GameEngine.getInstance()
        val shader = shaderProgram
        val displacement = d
        if (shader != null && displacement?.screenBase != null) {
            shader.a("screenBase", displacement.screenBase)
            shader.b("screenBaseSize", displacement.screenBase)
            shader.a("u_resolution", gameEngine.screenWidth, gameEngine.screenHeight)
            shader.a("u_offsetBy", displacement.offsetBy)
            shader.a("u_uiScaling", 1f)
        }
        gameEngine.renderGraphicsEngine.b(f, e)
        gameEngine.renderGraphicsEngine.b(a, 0f, 0f, c)
    }

    companion object {
        @JvmStatic
        fun postBase(): FrameBufferHelper = FrameBufferHelper("assets/shaders/post_base.frag")

        @JvmStatic
        fun postDisplacement(): FrameBufferHelper =
            FrameBufferHelper("assets/shaders/post_displacement.frag").apply {
                d = KoolDisplacementEffect()
                c.a(d)
            }
    }
}
